src/Microsoft.Tye.Hosting/ProcessRunner.cs

```diff
@@ -298,13 +298,16 @@ async Task RunApplicationAsync(IEnumerable<(int ExternalPort, int Port, string?
 
                                 WriteReplicaToStore(pid.ToString());
                                 service.ReplicaEvents.OnNext(new ReplicaEvent(ReplicaState.Started, status));
+                            },
+                            Build = async () =>
+                            {
+                                var buildResult = await ProcessUtil.RunAsync("dotnet", $"build \"{service.Status.ProjectFilePath}\" /nologo", throwOnError: false, workingDirectory: application.ContextDirectory);
```

Comment:
What happens on failure? Watch has this state machine to handle that case. Is that handled here?